src/snowflake/snowpark/mock/_connection.py

```diff
@@ -208,6 +215,47 @@ def __init__(self) -> None:
             "_PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING": True,
             "_PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING": True,
         }
+        self._connection_uuid = str(uuid.uuid4())
+        # by default, usage telemetry is collected
+        self._disable_local_testing_telemetry = (
+            options.get("disable_local_testing_telemetry", False) if options else False
+        )
+        self._oob_telemetry = LocalTestOOBTelemetryService.get_instance()
+        if self._disable_local_testing_telemetry:
+            # after disabling, the log will basically be a no-op, not sending any telemetry
+            self._oob_telemetry.disable()
+        else:
+            self._oob_telemetry.log_session_creation(self._connection_uuid)
+
+    def _log_not_supported_error(
```

Comment:
Starting this method with a _ implies it's a private method to MockServerConnection, but it's used in many places outside of this class. I think it would be better to remove the _ and have it be a public method.

Reply:
yes you're right, updated it to be a public method